File: app/src/main/java/com/pavel/a692group/UserAdapter.java
package com.pavel.a692group;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.pavel.a692group.room.entity.User;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Pavel Mazhnik on 03.03.19.
 * Description:
 * адаптер для отображния списка пользователей в GridView в EditDbActivity
 */

public class UserAdapter extends BaseAdapter {
    private Context mContext;
    private List<User> mUsers;

    public UserAdapter(Context context) {
        mContext = context;
        mUsers = new ArrayList<>();
    }

    public UserAdapter(Context context, List<User> users) {
        mContext = context;
        mUsers = users;
    }

    public void setData(List<User> user){
        mUsers = user;
    }

    public List<User> getData(){
        return mUsers;
    }

    public void updateUserFromPos(int position, User user){
        mUsers.get(position).updateUser(user);
    }

    @Override
    public int getCount() {
        return mUsers.size();
    }

    @Override
    public Object getItem(int position) {
        return mUsers.get(position);
    }

    @Override
    public long getItemId(int position) {
        return mUsers.get(position).getId();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = new View(mContext);
            LayoutInflater inflater = (LayoutInflater) mContext.getSystemService( Context.LAYOUT_INFLATER_SERVICE );
            convertView = inflater.inflate(R.layout.user_item, parent, false);

            ViewHolder viewHolder = new ViewHolder();
            viewHolder.Text_name =
                    (TextView)convertView.findViewById(R.id.item_user_name);
            viewHolder.Text_group =
                    (TextView)convertView.findViewById(R.id.item_user_group);
            // Store results of findViewById
            convertView.setTag(viewHolder);
        }
        TextView Text_name = ((ViewHolder)convertView.getTag()).Text_name;
        TextView Text_group = ((ViewHolder)convertView.getTag()).Text_group;
        Text_name.setText(mUsers.get(position).getName());
        Text_group.setText(mUsers.get(position).getGroup());
        return (convertView);
    }

    static class ViewHolder{
        TextView Text_name;
        TextView Text_group;
    }
}

File: app/src/main/java/com/pavel/a692group/presentation/users/UserAdapter.kt
package com.pavel.a692group.presentation.users

import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.lifecycle.LifecycleOwner
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.pavel.a692group.databinding.UserItemBinding
import com.pavel.a692group.data.entity.User

/**
 * Created by Pavel Mazhnik on 03.03.19.
 * Description:
 * адаптер для отображния списка пользователей в RecyclerView в EditDbActivity
 */
class UserAdapter(private val viewLifecycleOwner: LifecycleOwner, private val vm: UsersViewModel) :
    ListAdapter<User, UserAdapter.UserListHolder>(
        UserDiffCallback()
    ) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserListHolder {
        Log.d("UserAdapter", "onCreateViewHolder")
        val binding =
            UserItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        binding.lifecycleOwner = viewLifecycleOwner
        return UserListHolder(
            vm,
            binding
        )
    }

    override fun onBindViewHolder(holder: UserListHolder, position: Int) {
        val user = getItem(position)
        Log.d("UserAdapter", user.toString())

        if (user != null) {
            holder.bind(user)
        } else {
            holder.clear()
        }
    }

    private class UserDiffCallback : DiffUtil.ItemCallback<User>() {
        override fun areItemsTheSame(v1: User, v2: User): Boolean {
            return v1.id == v2.id
        }

        override fun areContentsTheSame(v1: User, v2: User): Boolean {
            return v1 == v2
        }
    }

    class UserListHolder(private val vm: UsersViewModel, private val binding: UserItemBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(user: User) {
            binding.user = user
            binding.usersViewModel = vm
            binding.executePendingBindings()
        }

        fun clear() {
        }
    }
}

